pkg/ccl/crosscluster/physical/stream_ingestion_frontier_processor.go

```diff
@@ -367,14 +368,14 @@ func (sf *streamIngestionFrontier) maybeUpdateProgress() error {
 		if err != nil {
 			return err
 		}
-		newProtectAbove := replicatedTime.Add(
-			-int64(replicationDetails.ReplicationTTLSeconds)*time.Second.Nanoseconds(), 0)
 
-		// If we have a CutoverTime set, keep the protected
-		// timestamp at or below the cutover time.
-		if !cutoverTime.IsEmpty() && cutoverTime.Less(newProtectAbove) {
-			newProtectAbove = cutoverTime
-		}
+		// No need to protect anything below replication start time.
+		replicationStartTime := md.Payload.GetStreamIngestion().ReplicationStartTime
+		newProtectAbove := replicationutils.ResolveHeartbeatTime(
+			replicatedTime, replicationStartTime, cutoverTime, replicationDetails.ReplicationTTLSeconds)
```

Comment:
nit: i think you can directly pass `streamProgress.CutoverTime` and get rid of the `cutoverTime` local var.